core/pom.xml

```diff
@@ -47,7 +47,7 @@
     <dependency>
       <groupId>com.google.code.gson</groupId>
       <artifactId>gson</artifactId>
-      <version>2.3.1</version>
+      <version>2.4</version>
```

Comment:
Can you check whether it is compatible with spark-2.3 also, as we will support spark-2.3 in near futreu.

Reply:
I think if gson ensures backward comparability, it is independent of spark version.